code/game/objects/items/storage/boxes/food_boxes.dm

```diff
@@ -335,49 +335,77 @@
 	foldable_result = null
 	custom_price = PAYCHECK_CREW
 
+	///Typepath of the type of gum that spawns with this box, this is passed to the wrapper for spawning in.
+	var/spawning_gum_type = /obj/item/food/bubblegum
+
 /obj/item/storage/box/gum/Initialize(mapload)
 	. = ..()
-	atom_storage.set_holdable(/obj/item/food/bubblegum)
+	atom_storage.set_holdable(/obj/item/storage/bubblegum_wrapper)
 	atom_storage.max_slots = 4
+	atom_storage.allow_big_nesting = TRUE
 
 /obj/item/storage/box/gum/PopulateContents()
 	for(var/i in 1 to 4)
-		new/obj/item/food/bubblegum(src)
+		new /obj/item/storage/bubblegum_wrapper(src, spawning_gum_type)
+
+/obj/item/storage/bubblegum_wrapper
+	name = "bubblegum wrapper"
+	icon = 'icons/obj/food/food.dmi'
+	icon_state = "bubblegum_wrapper"
+	w_class = WEIGHT_CLASS_TINY
+	resistance_flags = FLAMMABLE
+	item_flags = NOBLUDGEON|SKIP_FANTASY_ON_SPAWN
+	grind_results = list(/datum/reagent/aluminium = 1)
+
+	///The typepath of the type of gum that will spawn in our PopulateContents,
+	///this is set in Initialize by the gum box if there is one.
+	var/gum_to_spawn = /obj/item/food/bubblegum
+
+/obj/item/storage/bubblegum_wrapper/Initialize(mapload, spawning_gum_type)
+	if(!isnull(spawning_gum_type))
+		gum_to_spawn = spawning_gum_type
+	. = ..()
+	atom_storage.set_holdable(/obj/item/food/bubblegum)
+	atom_storage.max_slots = 1
+	update_appearance(UPDATE_OVERLAYS)
+
+/obj/item/storage/bubblegum_wrapper/PopulateContents()
+	new gum_to_spawn(src)
+
+/obj/item/storage/bubblegum_wrapper/update_overlays()
+	. = ..()
+	var/obj/item/food/bubblegum/gum_inside = locate() in contents
+	if(isnull(gum_inside))
+		return .
+	var/mutable_appearance/gum_overlay = mutable_appearance(gum_inside.icon, gum_inside.icon_state, layer = src.layer - 0.01)
```

Comment:
Should be using underlays instead

Reply:
I used underlay at first but replaced it cause we don't support it like we do overlays, this is what we do instead

Reply:
We do have full support for underlays, you'd just have to manually add/remove it from the list instead of adding it to update_overlays' return value. It would work, and this isn't a blocking comment, but I believe that here it'd be preferred.